### ISSUE TITLE
Remotecontrol/botowners

### ISSUE THREAD

**Issue by GoldenDarknessTLR** [ORelio/Minecraft-Console-Client#238]
Hey,
could you help me pls...What i must do that the bot autoaccepts tpa from specific players ?
I'm seraching some hours and testing, but i dont get it work... D:

**Comment by ORelio** [comment]
Hello,

You need to set bot owners in your INI file, as stated in [README](https://github.com/ORelio/Minecraft-Console-Client/blob/69542872d067d3b8a59a78303aca94318797c496/MinecraftClient/config/README.txt#L209).
````INI
botowners=GoldenDarknessTLR,ORelio #this is just an example
````
Of course you need to enable RemoteControl and enable auto tp accept.
````INI
[RemoteControl]
enabled=true
autotpaccept=true
tpaccepteveryone=false
````
finally, as stated in [README](https://github.com/ORelio/Minecraft-Console-Client/blob/69542872d067d3b8a59a78303aca94318797c496/MinecraftClient/config/README.txt#L210), chat messages need to be properly detected by MCC in order to react. If MCC does not know how teleport requests look like in chat, it cannot respond to them. ([Detecting chat messages](https://github.com/ORelio/Minecraft-Console-Client/blob/69542872d067d3b8a59a78303aca94318797c496/MinecraftClient/config/README.txt#L160)).

Simply put, if your teleport requests are not formatted like this:
````
PlayerName has requested to teleport to you.
````
Then you'll need to specify chat format in your INI file:
````INI
[ChatFormat]
builtins=false
public=^<([a-zA-Z0-9_]+)> (.+)$ #uncomment and adapt if necessary
private=^([a-zA-Z0-9_]+) whispers to you: (.+)$ #vanilla example
tprequest=^([a-zA-Z0-9_]+) has requested (?:to|that you) teleport to (?:you|them).
````
You can test your patterns at https://regex101.com/ ;)

**Comment by GoldenDarknessTLR** [comment]
Thanks, one thing i still have... the bot not react to my privat messages.
I think I must change the [Chatformat]  private=^([a-zA-Z0-9_]+) whispers to you: (.+)$
but still not get it to work.
If it could help you, in minecraft it looks so when i recieve a msg: http://prnt.sc/dvxsbr (blue is my name)

Hope you can help me with this & I love your Work 👍

**Comment by ORelio** [comment]
I assume the blue rectangle is the username. If so:
````INI
private=^\[(?:\[[^ ]+\] )([a-zA-Z0-9_]+) -> (?:me|mir)\] (.+)$
````
Try it: https://regex101.com/r/dBoQHF/1

**Comment by GoldenDarknessTLR** [comment]
Uhm, the bot not react to private messages with this code :(

And then one more thing... one of the botowners have a nick in game (~Name} and it not autoaccept tpa's from him.

**Comment by ORelio** [comment]
Technically the regex matches the sample you provided so I can't help much more. Just make sure builtins are disabled, and your regex is not commented with a `#` before `private=`.

Regarding `~` simply add an optional `~` character in front of the username:

````
private=^\[(?:\[[^ ]+\] )?~?([a-zA-Z0-9_]+) -> (?:me|mir)\] (.+)$
````
Try it: https://regex101.com/r/dBoQHF/5

**Comment by ORelio** [comment]
Seems like this issue is now resolved, feel free to reopen if needed :)

**Comment by Chtholly** [comment]
Hello, 
i have the same problem, my bot not respond to public & private messages by botowners.
[Ma'am][Elite✔]Chtholly_Nota: .
[[Ma'am][Mod]~Chtholly -> me] message...

ps. the bot dont accept my tparequest, but at Kirisaki it works... im using:
botowners=Chtholly_Nota,~Chtholly,Kirisaki
tprequest=^([a-zA-Z0-9_]+) has requested (?:to|that you) teleport to (?:you|them).

**Comment by ORelio** [comment]
@Chtholly Please follow the instruction above to build a regex for your own chat format.
Regarding teleport request you should use this:
`tprequest=~?^([a-zA-Z0-9_]+) has requested (?:to|that you) teleport to (?:you|them).`
... this way the leading `~` is stripped out and you can put `Chtholly` in your bot owner list.
